kartothek/core/dataset.py

```diff
@@ -133,13 +117,19 @@ def primary_indices_loaded(self) -> bool:
         return True
 
     @property
-    def tables(self) -> List[str]:
-        if self.table_meta:
-            return list(self.table_meta.keys())
+    def table_name(self) -> str:
+        if self._table_name:
+            return self._table_name
```

Comment:
`self._table_name` is set in __init__ and with a default value which evaluates to True.
Is it really required to keep the 3 cases, I would suggest to drop them.

Reply:
While the default is a str, it is actually possible to instantiate the class with `None` as a value. 

```python
In [1]: from kartothek.core.dataset import DatasetMetadata

In [2]: ds = DatasetMetadata("uuid")

In [3]: ds.table_name
Out[3]: 'table'

In [4]: ds = DatasetMetadata("uuid", table_name=None)

In [5]: ds
Out[5]: DatasetMetadata(uuid=uuid, table_name=<Unknown Table>, partition_keys=[], metadata_version=4, indices=[], explicit_partitions=True)

In [6]: ds.table_name
Out[6]: '<Unknown Table>'
```

This can happen in the edge case where one tries to read an empty dataset. The problem arises because we do not persist the tables as part of the JSON dataset metadata but is usually inferred by parsing the keys of the partitions. The edge case of an entirely empty dataset (i.e. no files) was not anticipated back then and we have multiple places in the test suite which test explicitly this case of not having any files. I do not recall the actual application for this, though. In favour of not breaking too much I would keep this as is, for now.